### PR TITLE
[contam visibility] : accounting for cases with 100 percent visibility

### PR DESCRIPTION
**Pull request by jaymedina**
aka , targets that will return an empty badPAs array.

### CODE REVIEW

**Comment by falkben** [comment]
Do you maybe want to include: https://github.com/ExoCTK/exoctk/pull/437?

File: exoctk/contam_visibility/visibilityPA.py [hunk 1]
```diff
@@ -405,17 +405,22 @@ def using_gtvt(
...
+        for idx in range(1, len(badPAs)):
+
+            if ((badPAs[idx - 1] + 1) == badPAs[idx]):
 
-        if ((badPAs[idx - 1] + 1) == badPAs[idx]):
+                grouped_badPAs[len(grouped_badPAs) - 1].append(badPAs[idx])
 
-            grouped_badPAs[len(grouped_badPAs) - 1].append(badPAs[idx])
+            elif ((badPAs[idx - 1] + 1) < badPAs[idx]):
+                grouped_badPAs.append([badPAs[idx]])
```
Comment (falkben) [inline exoctk/contam_visibility/visibilityPA.py:419]: As a list comprehension :)

```python
import itertools
grouped_badPAs = [[e[0] for e in d[1]] for d in itertools.groupby(
                  enumerate(badPAs.tolist()), key=lambda i_x: i_x[0] - i_x[1])]
```

We use [`itertools.groupby`](https://docs.python.org/3/library/itertools.html#itertools.groupby) to do the heavy lifting here. We use enumerate and a lambda function to do the grouping.  When `value - index` (from the enumerate) changes from one item to the next, we have a break. `groupby` returns an iterator of the group index and the group itself as an iterator. We can then iterate over these groups to create the new list of lists.

**Comment by falkben** [comment]
this fixes the issue i was having, thanks

**Review by nespinoza — COMMENTED** [review]
Hi @jaymedina,

 This works like a charm! Just tested it and solves both the bug on the visibility and contamination overlap tools. One detail before I approve this change: could you please change the versioning so we account for these bug-fixes on the ExoCTK version? This should be version 1.1.1. This version number change should be done in the `setup.py` file.  I'll create an accompanying CHANGELOG.MD file to keep track of these.

Thanks!
Néstor

**Comment by nespinoza** [comment]
Oh, and also please include #437!

File: exoctk/contam_visibility/visibilityPA.py (hunk 1 above)
Comment (jaymedina) [reply exoctk/contam_visibility/visibilityPA.py:419]: This is great! I think you might have also solved an optimization problem I was having... thanks! I'll be making a future PR implementing `itertools` for list comprehension. @falkben

**Comment by jaymedina** [comment]
@nespinoza this is ready to be merged if the code looks OK with you. I will work with Ben to get PR #437 merged separately. 

Thanks

**Review by hover2pi — APPROVED** [review]
Looks good to go. Thanks Jenny!

**Pull request merged**